src/main/java/seedu/recipe/model/recipe/IngredientBuilder.java

```diff
@@ -0,0 +1,68 @@
+package seedu.recipe.model.recipe;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.recipe.commons.util.AppUtil.checkArgument;
+import static seedu.recipe.model.recipe.ingredient.IngredientParser.NAME_PREFIX;
+import static seedu.recipe.model.recipe.ingredient.IngredientParser.parse;
+
+import java.util.HashMap;
+import java.util.List;
+
+import seedu.recipe.logic.parser.Prefix;
+
+/**
+ * Represents a recipe's ingredient in the recipe book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidIngredient(String)}
+ */
+public class IngredientBuilder {
+    public static final String MESSAGE_CONSTRAINTS =
+            "Ingredients should follow this format: \n"
+            + "`-a {AMOUNT} -e {ESTIMATED AMOUNT} -n {NAME} `"
+            + "`-cn {COMMON NAME} [-r {REMARKS}] [-s {SUBSTITUTION}]\n"
```

Comment:
I feel that common name could be a potentially removeable field (or at least optional) since a user would already fill the common name under the `-n` field for their own understanding

Reply:
Same goes for the `-e` estimated amount field